agent/src/agent/pipeline/manager.py

```diff
@@ -124,6 +127,13 @@ def reset_pipeline_retries(pipeline_: Pipeline):
         pipeline.repository.save(pipeline_.retries)
 
 
+def reset_pipeline_notifications(pipeline_: Pipeline):
+    if pipeline_.notifications:
+        if pipeline_.notifications.no_data_notification:
```

Comment:
Why not a single line if?
`if pipeline_.notifications and pipeline_.notifications.no_data_notification:`